From 38cf0dac298ab781fa89ce9cd00bf5236efbe629 Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Thu, 12 Apr 2012 01:51:49 +0200
Subject: [PATCH] move more common files to shared/ and add them to shared.la
 (cherry picked from commit
 f33d3ec1d7521c91da8b30ad5cb345d6416bb07d)

Conflicts:

	Makefile.am
---
 Makefile.am                                 |  159 ++++++++++-----------------
 src/{ => shared}/cgroup-show.c              |    0
 src/{ => shared}/cgroup-show.h              |    0
 src/{ => shared}/dbus-common.c              |    0
 src/{ => shared}/dbus-common.h              |    0
 src/{ => shared}/hostname-setup.c           |    0
 src/{ => shared}/hostname-setup.h           |    0
 src/{ => shared}/install.c                  |    0
 src/{ => shared}/install.h                  |    0
 src/{ => shared}/logs-show.c                |    0
 src/{ => shared}/logs-show.h                |    0
 src/{ => shared}/loopback-setup.c           |    0
 src/{ => shared}/loopback-setup.h           |    0
 src/{ => shared}/machine-id-setup.c         |    0
 src/{ => shared}/machine-id-setup.h         |    0
 src/{ => shared}/mount-setup.c              |    0
 src/{ => shared}/mount-setup.h              |    0
 src/{ => shared}/path-lookup.c              |    0
 src/{ => shared}/path-lookup.h              |    0
 src/{ => shared}/spawn-ask-password-agent.c |    0
 src/{ => shared}/spawn-ask-password-agent.h |    0
 src/{ => shared}/spawn-polkit-agent.c       |    0
 src/{ => shared}/spawn-polkit-agent.h       |    0
 src/{ => shared}/specifier.c                |    0
 src/{ => shared}/specifier.h                |    0
 src/{ => shared}/umount.c                   |    0
 src/{ => shared}/umount.h                   |    0
 src/{ => shared}/unit-name.c                |    0
 src/{ => shared}/unit-name.h                |    0
 src/{ => shared}/utmp-wtmp.c                |    0
 src/{ => shared}/utmp-wtmp.h                |    0
 src/{ => shared}/watchdog.c                 |    0
 src/{ => shared}/watchdog.h                 |    0
 33 files changed, 61 insertions(+), 98 deletions(-)
 rename src/{ => shared}/cgroup-show.c (100%)
 rename src/{ => shared}/cgroup-show.h (100%)
 rename src/{ => shared}/dbus-common.c (100%)
 rename src/{ => shared}/dbus-common.h (100%)
 rename src/{ => shared}/hostname-setup.c (100%)
 rename src/{ => shared}/hostname-setup.h (100%)
 rename src/{ => shared}/install.c (100%)
 rename src/{ => shared}/install.h (100%)
 rename src/{ => shared}/logs-show.c (100%)
 rename src/{ => shared}/logs-show.h (100%)
 rename src/{ => shared}/loopback-setup.c (100%)
 rename src/{ => shared}/loopback-setup.h (100%)
 rename src/{ => shared}/machine-id-setup.c (100%)
 rename src/{ => shared}/machine-id-setup.h (100%)
 rename src/{ => shared}/mount-setup.c (100%)
 rename src/{ => shared}/mount-setup.h (100%)
 rename src/{ => shared}/path-lookup.c (100%)
 rename src/{ => shared}/path-lookup.h (100%)
 rename src/{ => shared}/spawn-ask-password-agent.c (100%)
 rename src/{ => shared}/spawn-ask-password-agent.h (100%)
 rename src/{ => shared}/spawn-polkit-agent.c (100%)
 rename src/{ => shared}/spawn-polkit-agent.h (100%)
 rename src/{ => shared}/specifier.c (100%)
 rename src/{ => shared}/specifier.h (100%)
 rename src/{ => shared}/umount.c (100%)
 rename src/{ => shared}/umount.h (100%)
 rename src/{ => shared}/unit-name.c (100%)
 rename src/{ => shared}/unit-name.h (100%)
 rename src/{ => shared}/utmp-wtmp.c (100%)
 rename src/{ => shared}/utmp-wtmp.h (100%)
 rename src/{ => shared}/watchdog.c (100%)
 rename src/{ => shared}/watchdog.h (100%)

diff --git a/Makefile.am b/Makefile.am
index abe2e7f..32db59c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -100,8 +100,8 @@ AM_CPPFLAGS = \
 	-DUSER_GENERATOR_PATH=\"$(usergeneratordir)\" \
 	-DSYSTEM_SHUTDOWN_PATH=\"$(systemshutdowndir)\" \
 	-DSYSTEMD_KBD_MODEL_MAP=\"$(pkgdatadir)/kbd-model-map\" \
-        -DX_SERVER=\"$(bindir)/X\" \
-        -DPOLKIT_AGENT_BINARY_PATH=\"$(bindir)/pkttyagent\" \
+	-DX_SERVER=\"$(bindir)/X\" \
+	-DPOLKIT_AGENT_BINARY_PATH=\"$(bindir)/pkttyagent\" \
 	-I $(top_srcdir)/src \
 	-I $(top_srcdir)/src/shared \
 	-I $(top_srcdir)/src/readahead \
@@ -570,14 +570,50 @@ libsystemd_shared_la_SOURCES = \
 	src/shared/socket-util.h \
 	src/shared/cgroup-util.c \
 	src/shared/cgroup-util.h \
+	src/shared/dbus-common.c \
+	src/shared/dbus-common.h \
+	src/shared/hostname-setup.c \
+	src/shared/hostname-setup.h \
+	src/shared/logs-show.c \
+	src/shared/logs-show.h \
+	src/shared/cgroup-show.c \
+	src/shared/cgroup-show.h \
+	src/shared/unit-name.c \
+	src/shared/unit-name.h \
+	src/shared/utmp-wtmp.c \
+	src/shared/utmp-wtmp.h \
+	src/shared/watchdog.c \
+	src/shared/watchdog.h \
+	src/shared/umount.c \
+	src/shared/umount.h \
+	src/shared/spawn-ask-password-agent.c \
+	src/shared/spawn-ask-password-agent.h \
+	src/shared/mount-setup.c \
+	src/shared/mount-setup.h \
+	src/shared/machine-id-setup.c \
+	src/shared/machine-id-setup.h \
+	src/shared/loopback-setup.h \
+	src/shared/loopback-setup.c \
+	src/shared/specifier.c \
+	src/shared/specifier.h \
+	src/shared/spawn-polkit-agent.c \
+	src/shared/spawn-polkit-agent.h \
 	src/shared/list.h \
 	src/shared/macro.h
 
+libsystemd_shared_la_CFLAGS = \
+	$(AM_CFLAGS) \
+	$(DBUS_CFLAGS)
+
 # ------------------------------------------------------------------------------
 noinst_LTLIBRARIES += \
 	libsystemd-label.la
 
 libsystemd_label_la_SOURCES = \
+	src/shared/install.c \
+	src/shared/install.h \
+	src/shared/path-lookup.c \
+	src/shared/path-lookup.h \
 	src/shared/cgroup-label.c \
 	src/shared/socket-label.c \
 	src/shared/label.c \
@@ -587,6 +623,7 @@ libsystemd_label_la_SOURCES = \
 
 libsystemd_label_la_CFLAGS = \
 	$(AM_CFLAGS) \
+	$(DBUS_CFLAGS) \
 	$(SELINUX_CFLAGS)
 
 libsystemd_label_la_LIBADD = \
@@ -642,28 +679,6 @@ noinst_LTLIBRARIES += \
 libsystemd_core_la_SOURCES = \
 	src/def.h \
 	src/missing.h \
-	src/dbus-common.c \
-	src/dbus-common.h \
-	src/watchdog.c \
-	src/watchdog.h \
-	src/loopback-setup.h \
-	src/loopback-setup.c \
-	src/hostname-setup.c \
-	src/hostname-setup.h \
-	src/specifier.c \
-	src/specifier.h \
-	src/install.c \
-	src/install.h \
-	src/path-lookup.c \
-	src/path-lookup.h \
-	src/unit-name.c \
-	src/unit-name.h \
-	src/utmp-wtmp.c \
-	src/utmp-wtmp.h \
-	src/machine-id-setup.c \
-	src/machine-id-setup.h \
-	src/mount-setup.c \
-	src/mount-setup.h \
 	src/linux/auto_dev-ioctl.h \
 	src/linux/fanotify.h \
 	src/core/unit.c \
@@ -829,15 +844,13 @@ test_ns_LDADD = \
 	libsystemd-core.la
 
 test_loopback_SOURCES = \
-	src/test-loopback.c \
-	src/loopback-setup.c
+	src/test-loopback.c
 
 test_loopback_LDADD = \
 	libsystemd-shared.la
 
 test_hostname_SOURCES = \
-	src/test-hostname.c \
-	src/hostname-setup.c
+	src/test-hostname.c
 
 test_hostname_LDADD = \
 	libsystemd-shared.la
@@ -863,17 +876,13 @@ test_env_replace_LDADD = \
 	libsystemd-shared.la
 
 test_strv_SOURCES = \
-	src/test-strv.c \
-	src/specifier.c
+	src/test-strv.c
 
 test_strv_LDADD = \
 	libsystemd-shared.la
 
 test_install_SOURCES = \
-	src/test-install.c \
-	src/install.c \
-	src/path-lookup.c \
-	src/unit-name.c
+	src/test-install.c
 
 test_install_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -884,17 +893,14 @@ test_install_LDADD = \
 	libsystemd-shared.la
 
 test_watchdog_SOURCES = \
-	src/test-watchdog.c \
-	src/watchdog.c \
-	src/watchdog.h
+	src/test-watchdog.c
 
 test_watchdog_LDADD = \
 	libsystemd-shared.la
 
 # ------------------------------------------------------------------------------
 systemd_initctl_SOURCES = \
-	src/initctl.c \
-	src/dbus-common.c
+	src/initctl.c
 
 systemd_initctl_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -907,9 +913,7 @@ systemd_initctl_LDADD = \
 
 # ------------------------------------------------------------------------------
 systemd_update_utmp_SOURCES = \
-	src/update-utmp.c \
-	src/dbus-common.c \
-	src/utmp-wtmp.c
+	src/update-utmp.c
 
 systemd_update_utmp_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -923,7 +927,6 @@ systemd_update_utmp_LDADD = \
 
 # ------------------------------------------------------------------------------
 systemd_shutdownd_SOURCES = \
-	src/utmp-wtmp.c \
 	src/shutdownd.c
 
 systemd_shutdownd_LDADD = \
@@ -936,12 +939,7 @@ pkginclude_HEADERS += \
 
 # ------------------------------------------------------------------------------
 systemd_shutdown_SOURCES = \
-	src/mount-setup.c \
-	src/umount.c \
-	src/umount.h \
-	src/shutdown.c \
-	src/watchdog.c \
-	src/watchdog.h
+	src/shutdown.c
 
 systemd_shutdown_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -974,7 +972,6 @@ systemd_tmpfiles_LDADD = \
 
 # ------------------------------------------------------------------------------
 systemd_machine_id_setup_SOURCES = \
-	src/machine-id-setup.c \
 	src/machine-id-main.c
 
 systemd_machine_id_setup_LDADD = \
@@ -991,8 +988,7 @@ systemd_sysctl_LDADD = \
 
 # ------------------------------------------------------------------------------
 systemd_fsck_SOURCES = \
-	src/fsck.c \
-	src/dbus-common.c
+	src/fsck.c
 
 systemd_fsck_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -1032,8 +1028,7 @@ systemd_detect_virt_LDADD = \
 
 # ------------------------------------------------------------------------------
 systemd_getty_generator_SOURCES = \
-	src/getty-generator.c \
-	src/unit-name.c
+	src/getty-generator.c
 
 systemd_getty_generator_LDADD = \
 	libsystemd-label.la \
@@ -1049,16 +1044,14 @@ systemd_rc_local_generator_LDADD = \
 
 # ------------------------------------------------------------------------------
 systemd_remount_api_vfs_SOURCES = \
-	src/remount-api-vfs.c \
-	src/mount-setup.c
+	src/remount-api-vfs.c
 
 systemd_remount_api_vfs_LDADD = \
 	libsystemd-shared.la
 
 # ------------------------------------------------------------------------------
 systemd_cgroups_agent_SOURCES = \
-	src/cgroups-agent.c \
-	src/dbus-common.c
+	src/cgroups-agent.c
 
 systemd_cgroups_agent_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -1070,20 +1063,7 @@ systemd_cgroups_agent_LDADD = \
 
 # ------------------------------------------------------------------------------
 systemctl_SOURCES = \
-	src/systemctl.c \
-	src/utmp-wtmp.c \
-	src/dbus-common.c \
-	src/path-lookup.c \
-	src/cgroup-show.c \
-	src/cgroup-show.h \
-	src/unit-name.c \
-	src/install.c \
-	src/spawn-ask-password-agent.c \
-	src/spawn-ask-password-agent.h \
-	src/spawn-polkit-agent.c \
-	src/spawn-polkit-agent.h \
-	src/logs-show.c \
-	src/logs-show.h
+	src/systemctl.c
 
 systemctl_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -1124,9 +1104,7 @@ systemd_reply_password_LDADD = \
 
 # ------------------------------------------------------------------------------
 systemd_cgls_SOURCES = \
-	src/cgls.c \
-	src/cgroup-show.c \
-	src/cgroup-show.h
+	src/cgls.c
 
 systemd_cgls_LDADD = \
 	libsystemd-shared.la
@@ -1140,13 +1118,12 @@ systemd_cgtop_LDADD = \
 
 # ------------------------------------------------------------------------------
 systemd_nspawn_SOURCES = \
-	src/nspawn.c \
-	src/loopback-setup.c
+	src/nspawn.c
 
 systemd_nspawn_LDADD = \
 	libsystemd-label.la \
-	libsystemd-shared.la \
 	libsystemd-capability.la \
+	libsystemd-shared.la \
 	libsystemd-daemon.la
 
 # ------------------------------------------------------------------------------
@@ -1205,8 +1182,7 @@ systemd_gnome_ask_password_agent_LDADD = \
 # ------------------------------------------------------------------------------
 systemd_tty_ask_password_agent_SOURCES = \
 	src/tty-ask-password-agent.c \
-	src/ask-password-api.c \
-	src/utmp-wtmp.c
+	src/ask-password-api.c
 
 systemd_tty_ask_password_agent_LDADD = \
 	libsystemd-label.la \
@@ -1392,9 +1368,7 @@ systemd_cat_LDADD = \
 	libsystemd-journal.la
 
 journalctl_SOURCES = \
-	src/journal/journalctl.c \
-	src/logs-show.c \
-	src/logs-show.h
+	src/journal/journalctl.c
 
 journalctl_LDADD = \
 	libsystemd-shared.la \
@@ -1776,8 +1750,7 @@ systemd_cryptsetup_LDADD = \
 	$(LIBCRYPTSETUP_LIBS)
 
 systemd_cryptsetup_generator_SOURCES = \
-	src/cryptsetup/cryptsetup-generator.c \
-	src/unit-name.c
+	src/cryptsetup/cryptsetup-generator.c
 
 systemd_cryptsetup_generator_LDADD = \
 	libsystemd-label.la \
@@ -1798,7 +1771,6 @@ endif
 if ENABLE_HOSTNAMED
 systemd_hostnamed_SOURCES = \
 	src/hostname/hostnamed.c \
-	src/dbus-common.c \
 	src/polkit.c
 
 systemd_hostnamed_CFLAGS = \
@@ -1849,7 +1821,6 @@ endif
 if ENABLE_LOCALED
 systemd_localed_SOURCES = \
 	src/locale/localed.c \
-	src/dbus-common.c \
 	src/polkit.c
 
 systemd_localed_CFLAGS = \
@@ -1911,7 +1882,6 @@ endif
 if ENABLE_TIMEDATED
 systemd_timedated_SOURCES = \
 	src/timedate/timedated.c \
-	src/dbus-common.c \
 	src/polkit.c
 
 systemd_timedated_CFLAGS = \
@@ -1975,7 +1945,6 @@ systemd_logind_SOURCES = \
 	src/login/logind-session-dbus.c \
 	src/login/logind-seat-dbus.c \
 	src/login/logind-user-dbus.c \
-	src/dbus-common.c \
 	src/dbus-loop.c \
 	src/polkit.c \
 	src/login/logind-acl.h
@@ -2016,12 +1985,7 @@ rootlibexec_PROGRAMS += \
 
 loginctl_SOURCES = \
 	src/login/loginctl.c \
-	src/login/sysfs-show.c \
-	src/dbus-common.c \
-	src/spawn-polkit-agent.c \
-	src/spawn-polkit-agent.h \
-	src/cgroup-show.c \
-	src/cgroup-show.h
+	src/login/sysfs-show.c
 
 loginctl_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -2064,8 +2028,7 @@ libsystemd_login_la_LIBADD = \
 
 if HAVE_PAM
 pam_systemd_la_SOURCES = \
-	src/login/pam-module.c \
-	src/dbus-common.c
+	src/login/pam-module.c
 
 pam_systemd_la_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -2139,7 +2102,7 @@ polkitpolicy_in_files += \
 logind-install-data-hook:
 	$(MKDIR_P) -m 0755 \
 		$(DESTDIR)$(systemunitdir)/multi-user.target.wants \
-                $(DESTDIR)$(localstatedir)/lib/systemd
+		$(DESTDIR)$(localstatedir)/lib/systemd
 	( cd $(DESTDIR)$(systemunitdir) && \
 		rm -f dbus-org.freedesktop.login1.service && \
 		$(LN_S) systemd-logind.service dbus-org.freedesktop.login1.service)
diff --git a/src/cgroup-show.c b/src/shared/cgroup-show.c
similarity index 100%
rename from src/cgroup-show.c
rename to src/shared/cgroup-show.c
diff --git a/src/cgroup-show.h b/src/shared/cgroup-show.h
similarity index 100%
rename from src/cgroup-show.h
rename to src/shared/cgroup-show.h
diff --git a/src/dbus-common.c b/src/shared/dbus-common.c
similarity index 100%
rename from src/dbus-common.c
rename to src/shared/dbus-common.c
diff --git a/src/dbus-common.h b/src/shared/dbus-common.h
similarity index 100%
rename from src/dbus-common.h
rename to src/shared/dbus-common.h
diff --git a/src/hostname-setup.c b/src/shared/hostname-setup.c
similarity index 100%
rename from src/hostname-setup.c
rename to src/shared/hostname-setup.c
diff --git a/src/hostname-setup.h b/src/shared/hostname-setup.h
similarity index 100%
rename from src/hostname-setup.h
rename to src/shared/hostname-setup.h
diff --git a/src/install.c b/src/shared/install.c
similarity index 100%
rename from src/install.c
rename to src/shared/install.c
diff --git a/src/install.h b/src/shared/install.h
similarity index 100%
rename from src/install.h
rename to src/shared/install.h
diff --git a/src/logs-show.c b/src/shared/logs-show.c
similarity index 100%
rename from src/logs-show.c
rename to src/shared/logs-show.c
diff --git a/src/logs-show.h b/src/shared/logs-show.h
similarity index 100%
rename from src/logs-show.h
rename to src/shared/logs-show.h
diff --git a/src/loopback-setup.c b/src/shared/loopback-setup.c
similarity index 100%
rename from src/loopback-setup.c
rename to src/shared/loopback-setup.c
diff --git a/src/loopback-setup.h b/src/shared/loopback-setup.h
similarity index 100%
rename from src/loopback-setup.h
rename to src/shared/loopback-setup.h
diff --git a/src/machine-id-setup.c b/src/shared/machine-id-setup.c
similarity index 100%
rename from src/machine-id-setup.c
rename to src/shared/machine-id-setup.c
diff --git a/src/machine-id-setup.h b/src/shared/machine-id-setup.h
similarity index 100%
rename from src/machine-id-setup.h
rename to src/shared/machine-id-setup.h
diff --git a/src/mount-setup.c b/src/shared/mount-setup.c
similarity index 100%
rename from src/mount-setup.c
rename to src/shared/mount-setup.c
diff --git a/src/mount-setup.h b/src/shared/mount-setup.h
similarity index 100%
rename from src/mount-setup.h
rename to src/shared/mount-setup.h
diff --git a/src/path-lookup.c b/src/shared/path-lookup.c
similarity index 100%
rename from src/path-lookup.c
rename to src/shared/path-lookup.c
diff --git a/src/path-lookup.h b/src/shared/path-lookup.h
similarity index 100%
rename from src/path-lookup.h
rename to src/shared/path-lookup.h
diff --git a/src/spawn-ask-password-agent.c b/src/shared/spawn-ask-password-agent.c
similarity index 100%
rename from src/spawn-ask-password-agent.c
rename to src/shared/spawn-ask-password-agent.c
diff --git a/src/spawn-ask-password-agent.h b/src/shared/spawn-ask-password-agent.h
similarity index 100%
rename from src/spawn-ask-password-agent.h
rename to src/shared/spawn-ask-password-agent.h
diff --git a/src/spawn-polkit-agent.c b/src/shared/spawn-polkit-agent.c
similarity index 100%
rename from src/spawn-polkit-agent.c
rename to src/shared/spawn-polkit-agent.c
diff --git a/src/spawn-polkit-agent.h b/src/shared/spawn-polkit-agent.h
similarity index 100%
rename from src/spawn-polkit-agent.h
rename to src/shared/spawn-polkit-agent.h
diff --git a/src/specifier.c b/src/shared/specifier.c
similarity index 100%
rename from src/specifier.c
rename to src/shared/specifier.c
diff --git a/src/specifier.h b/src/shared/specifier.h
similarity index 100%
rename from src/specifier.h
rename to src/shared/specifier.h
diff --git a/src/umount.c b/src/shared/umount.c
similarity index 100%
rename from src/umount.c
rename to src/shared/umount.c
diff --git a/src/umount.h b/src/shared/umount.h
similarity index 100%
rename from src/umount.h
rename to src/shared/umount.h
diff --git a/src/unit-name.c b/src/shared/unit-name.c
similarity index 100%
rename from src/unit-name.c
rename to src/shared/unit-name.c
diff --git a/src/unit-name.h b/src/shared/unit-name.h
similarity index 100%
rename from src/unit-name.h
rename to src/shared/unit-name.h
diff --git a/src/utmp-wtmp.c b/src/shared/utmp-wtmp.c
similarity index 100%
rename from src/utmp-wtmp.c
rename to src/shared/utmp-wtmp.c
diff --git a/src/utmp-wtmp.h b/src/shared/utmp-wtmp.h
similarity index 100%
rename from src/utmp-wtmp.h
rename to src/shared/utmp-wtmp.h
diff --git a/src/watchdog.c b/src/shared/watchdog.c
similarity index 100%
rename from src/watchdog.c
rename to src/shared/watchdog.c
diff --git a/src/watchdog.h b/src/shared/watchdog.h
similarity index 100%
rename from src/watchdog.h
rename to src/shared/watchdog.h
